gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/services/observability/observabilty_service.py

```diff
@@ -45,13 +46,20 @@ def check_observability_setting(setting: ObservabilitySetting) -> bool:
     return get_callback_handler_factory(setting).check_observability_setting()
 
 
-def get_observability_info(observability_handler) -> Optional[ObservabilityInfo]:
+def get_observability_info(observability_handler, trace_name: Optional[str] = None) -> Optional[ObservabilityInfo]:
     """Get the observability Information"""
-    if isinstance(observability_handler, LangfuseCallbackHandler):
-        return ObservabilityInfo(
-            trace_id=observability_handler.trace.id,
-            trace_name=observability_handler.trace_name,
-            trace_url=observability_handler.get_trace_url()
-        )
-    else:
-        return None
\ No newline at end of file
+    if not isinstance(observability_handler, LangfuseCallbackHandler):
+        return None
+
+    trace_id = getattr(observability_handler, 'last_trace_id', None)
+    if trace_id is None:
+        return None
+
+    langfuse_client = get_client()
```

Comment:
Are you sure you are getting the right client here ?
You should be using the client public key to be sure at least.


This could lead to rights access errors.